Tidy adjoint tests (#368)
ecmwf-ifs/ectrans: tests/trans/adjoint/CMakeLists.txt

# granted to it by virtue of its status as an intergovernmental organisation
# nor does it submit to any jurisdiction.

# Tests for adjoint functionality

foreach( precision ${precisions} )

  ecbuild_add_library(
    TARGET           adjoint_utils_${precision}
    LINKER_LANGUAGE  Fortran
    SOURCES          utils.F90
    PUBLIC_LIBS      fiat trans_${precision}
  )
  ecbuild_target_fortran_module_directory(
    TARGET adjoint_utils_${precision}
    MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/tests/trans/adjoint/module/adjoint_utils_${precision}
  )

  foreach( platform ${platforms} )

    if( "${platform}" MATCHES "cpu" )
      set( platform_tag "" )
    endif()
    if( "${platform}" MATCHES "gpu" )
      set( platform_tag "_gpu" )
    endif()

    foreach( mpi ${ntasks} )
      # Add a test for tangent-linear/adjoint correspondence of INV_TRANS and DIR_TRANS combined
      ecbuild_add_test(TARGET ectrans_test_adjoint_${platform}_mpi${mpi}_${precision}
        SOURCES test_adjoint.F90
        LIBS trans${platform_tag}_${precision} parkind_${precision} adjoint_utils_${precision}
        LINKER_LANGUAGE Fortran
        MPI ${mpi}
        OMP 1
      )

>       # Add tests for tangent-linear/adjoint correspondence of INV_TRANS and DIR_TRANS

Author: ```suggestion
      # Add test for tangent-linear/adjoint correspondence of DIR_TRANS only
```

      ecbuild_add_test(TARGET ectrans_test_dirtrans_adjoint_${platform}_mpi${mpi}_${precision}
        SOURCES test_dirtrans_adjoint.F90
        LIBS trans${platform_tag}_${precision} parkind_${precision} adjoint_utils_${precision}
        LINKER_LANGUAGE Fortran
        MPI ${mpi}
        OMP 1
      )

>       ecbuild_add_test(TARGET ectrans_test_invtrans_adjoint_${platform}_mpi${mpi}_${precision}

Author: ```suggestion
      # Add test for tangent-linear/adjoint correspondence of INV_TRANS only
      ecbuild_add_test(TARGET ectrans_test_invtrans_adjoint_${platform}_mpi${mpi}_${precision}
```
